File: crates/ecaa-conformance/tests/fixtures/minimal-package-real/runtime/plotting_r/stages/quantification.R
# R-side quantification stage: peptide_coverage, ridgeline.
# Mirrors lib/plotting/stages/quantification.py.
#
# Plan reference: §S13.6 — proteomics quantification figures.

if (!exists("ecaa_register_figure")) {
  stop("source runtime/plotting_r/core.R before this stage module")
}

ecaa_register_figure("quantification", "peptide_coverage", function(ctx) {
  p <- .ecaa_manifest_path(ctx$manifest, ctx$outputs_dir, "coverage_table")
  if (is.null(p)) stop("manifest.coverage_table required")
  df <- .ecaa_load_tsv(p, list(
    position = c("position", "residue", "aa_pos"),
    coverage = c("coverage", "depth", "n_peptides")
  ))
  if (is.null(df)) stop(sprintf("unparseable coverage table: %s", p))
  ecaa_peptide_coverage_r(df, title = "Peptide coverage")
})

ecaa_register_figure("quantification", "ridgeline", function(ctx) {
  p <- .ecaa_manifest_path(ctx$manifest, ctx$outputs_dir, "intensity_table")
  if (is.null(p)) stop("manifest.intensity_table required")
  df <- .ecaa_load_tsv(p, list(
    group = c("group", "sample", "run", "channel"),
    value = c("value", "intensity", "log_intensity", "abundance")
  ))
  if (is.null(df)) stop(sprintf("unparseable intensity table: %s", p))
  ecaa_ridgeline_r(df, title = "Ion-intensity ridgeline")
})

File: testdata/emitted-packages/12-stoeckius-cbmc-citeseq/runtime/plotting/stages/quantification.py
"""Quantification stage — proteomics taxonomy. Renders peptide
coverage of the target proteins and a ridgeline plot of ion-intensity
distributions across samples.

Plan reference: §S13 (proteomics taxonomy stage `quantification`).

Manifest contract:
- ``coverage_table``: TSV with `position`, `coverage`. Optional
  `protein` column when multiple targets — when present, the first
  protein is rendered (extension to multi-protein follow-on).
- ``intensity_table``: TSV with `group`, `value`. Group = sample id;
  value = log-ion-intensity per peptide.
"""

from __future__ import annotations

from pathlib import Path
from typing import Optional

from ..core import (
    FigureContext,
    peptide_coverage,
    register_figure,
    ridgeline,
    stage_registry,
)
from ._shared import load_tsv_columns, manifest_path

FIGURES = stage_registry("quantification")


@register_figure(FIGURES, "peptide_coverage")
def peptide_coverage_fig(ctx: FigureContext, out: Path) -> Optional[Path]:
    p = manifest_path(ctx.manifest, ctx.outputs_dir, "coverage_table")
    if p is None:
        raise FileNotFoundError("manifest.coverage_table required")
    cols = load_tsv_columns(
        p,
        {
            "position": ("position", "residue", "aa_pos"),
            "coverage": ("coverage", "depth", "n_peptides"),
        },
    )
    if cols is None:
        raise FileNotFoundError(f"unparseable coverage table: {p}")
    return peptide_coverage(
        frame=cols, title="Peptide coverage", out=out
    )


@register_figure(FIGURES, "ridgeline")
def ridgeline_fig(ctx: FigureContext, out: Path) -> Optional[Path]:
    p = manifest_path(ctx.manifest, ctx.outputs_dir, "intensity_table")
    if p is None:
        raise FileNotFoundError("manifest.intensity_table required")
    cols = load_tsv_columns(
        p,
        {
            "group": ("group", "sample", "run", "channel"),
            "value": ("value", "intensity", "log_intensity", "abundance"),
        },
    )
    if cols is None:
        raise FileNotFoundError(f"unparseable intensity table: {p}")
    return ridgeline(
        frame=cols, title="Ion-intensity ridgeline", out=out
    )
